### PR TITLE
Added Raven DB Support

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

File: src/HotChocolate/Core/src/Execution/DependencyInjection/SchemaRequestExecutorBuilderExtensions.Types.cs [hunk 1]
```diff
@@ -1263,6 +1263,38 @@ public static partial class SchemaRequestExecutorBuilderExtensions
...
+    /// <param name="namedType">
+    /// The GraphQL type.
+    /// </param>
+    /// <returns>
+    /// Returns the GraphQL configuration builder.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="builder"/> or <paramref name="builder"/> is <c>null</c>
+    /// </exception>
+    public static IRequestExecutorBuilder AddType(
```
Comment (michaelstaib) [inline src/HotChocolate/Core/src/Execution/DependencyInjection/SchemaRequestExecutorBuilderExtensions.Types.cs:1281]: this should be `AddTypeExtensiosn`

File: src/HotChocolate/Core/src/Types.OffsetPagination/Extensions/CursorPaginationResolverContextExtensions.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 // ReSharper disable once CheckNamespace
 namespace HotChocolate.Types.Pagination;
 
-internal static class CursorPaginationResolverContextExtensions
+internal static class OffsetPaginationResolverContextExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Core/src/Types.OffsetPagination/Extensions/CursorPaginationResolverContextExtensions.cs:6]: The file name needs to be alligned.

File: src/HotChocolate/Data/src/Data/Filters/Expressions/QueryableFilterProvider.cs [hunk 3]
```diff
@@ -133,6 +126,18 @@ private ApplyFiltering CreateApplicator<TEntityType>(string argumentName)
...
     }
 
+    protected virtual object? ApplyToResult<TEntityType>(
+        object? input,
+        Expression<Func<TEntityType, bool>> where)
+        => input switch
+        {
+            IQueryable<TEntityType> q => q.Where(where),
+            IEnumerable<TEntityType> e => e.AsQueryable().Where(where),
+            QueryableExecutable<TEntityType> ex =>
```
Comment (michaelstaib) [inline src/HotChocolate/Data/src/Data/Filters/Expressions/QueryableFilterProvider.cs:136]: ex is for exceptions ... qe?

File: src/HotChocolate/Data/test/Data.Projections.SqlServer.Tests/__snapshots__/QueryableFirstOrDefaultTests.Create_DeepFilterObjectTwoProjections_Executable_NET7_0.snap [hunk 4]
```diff
@@ -16,4 +18,10 @@
...
     "rootExecutable": null
   }
-}
\ No newline at end of file
+}
+---------------
+
+SQL:
+---------------
+Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1[HotChocolate.Data.Projections.QueryableFirstOrDefaultTests+Bar]
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Projections.SqlServer.Tests/__snapshots__/QueryableFirstOrDefaultTests.Create_DeepFilterObjectTwoProjections_Executable_NET7_0.snap:26]: is this correct? shouldn't this be SQL or something more readable?

File: src/HotChocolate/Raven/src/Data/Pagination/RavenCursorPagingHandler.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+using HotChocolate.Resolvers;
+using HotChocolate.Types.Pagination;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven.Pagination;
+
+internal class RavenCursorPagingHandler<TEntity> : CursorPagingHandler
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Pagination/RavenCursorPagingHandler.cs:9]: sealed?

File: src/HotChocolate/Raven/src/Data/Pagination/RavenCursorPagingProvider.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
+using System.Reflection;
+using HotChocolate.Internal;
+using HotChocolate.Types.Pagination;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven.Pagination;
+
+public class RavenCursorPagingProvider : CursorPagingProvider
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Pagination/RavenCursorPagingProvider.cs:9]: sealed?

File: src/HotChocolate/Raven/src/Data/Pagination/RavenOffsetPagingHandler.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
+using HotChocolate.Resolvers;
+using HotChocolate.Types.Pagination;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven.Pagination;
+
+internal class RavenOffsetPagingHandler<TEntity> : OffsetPagingHandler
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Pagination/RavenOffsetPagingHandler.cs:9]: sealed?

File: src/HotChocolate/Raven/src/Data/Pagination/RavenOffsetPagingProvider.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+using HotChocolate.Types.Pagination;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven.Pagination;
+
+/// <summary>
+/// An offset paging provider for Raven that create pagination queries
+/// </summary>
+public class RavenOffsetPagingProvider : OffsetPagingProvider
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Pagination/RavenOffsetPagingProvider.cs:12]: sealed?

File: src/HotChocolate/Raven/src/Data/Pagination/RavenPagingContainer.cs [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        return this;
+    }
+
+    public RavenPagingContainer<TEntity> Take(int take)
+    {
+        _query = _query.Take(take);
+        return this;
+    }
+
+    private sealed class TaskCache
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Pagination/RavenPagingContainer.cs:126]: this is not a task cache ... more something that makes sure you only resolve something once. Task cache for dataloader for instance in fact caches tasks.

Comment (michaelstaib) [reply src/HotChocolate/Raven/src/Data/Pagination/RavenPagingContainer.cs:126]: it looks more like a Lazy

File: src/HotChocolate/Raven/src/Data/Pagination/RavenCursorPagingProvider.cs (hunk 6 above)
Comment (michaelstaib) [reply src/HotChocolate/Raven/src/Data/Pagination/RavenCursorPagingProvider.cs:9]: also do xml summary

File: src/HotChocolate/Raven/src/Data/Pagination/RavenOffsetPagingProvider.cs (hunk 8 above)
Comment (michaelstaib) [reply src/HotChocolate/Raven/src/Data/Pagination/RavenOffsetPagingProvider.cs:12]: also do xml summary

File: src/HotChocolate/Raven/src/Data/Projections/Extensions/RavenProjectionsRequestExecutorBuilderExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,11 @@
+using HotChocolate.Execution.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HotChocolate.Data;
+
+public static class RavenProjectionsRequestExecutorBuilderExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Projections/Extensions/RavenProjectionsRequestExecutorBuilderExtensions.cs:6]: also do xml summary

File: src/HotChocolate/Raven/src/Data/Projections/Extensions/RavenProjectionsSchemaBuilderExtensions.cs [hunk 11]
```diff
@@ -0,0 +1,7 @@
+namespace HotChocolate.Data;
+
+public static class RavenProjectionsSchemaBuilderExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Projections/Extensions/RavenProjectionsSchemaBuilderExtensions.cs:3]: also do xml summary

File: src/HotChocolate/Raven/src/Data/Projections/RavenQueryableProjectionProvider.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
+using System.Linq.Expressions;
+using HotChocolate.Data.Projections;
+using HotChocolate.Data.Projections.Expressions;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven.Projections;
+
+public class RavenQueryableProjectionProvider : QueryableProjectionProvider
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Projections/RavenQueryableProjectionProvider.cs:10]: sealed?

also do xml summary

File: src/HotChocolate/Raven/src/Data/RavenAsyncDocumentQueryExecutable.cs [hunk 13]
```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven;
+
+/// <summary>
+/// Is the base class for a executable for the Raven.
+/// </summary>
+public class RavenAsyncDocumentQueryExecutable<T>
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/RavenAsyncDocumentQueryExecutable.cs:9]: sealed?

File: src/HotChocolate/Raven/src/Data/RavenExecutableExtensions.cs [hunk 14]
```diff
@@ -0,0 +1,36 @@
+using HotChocolate.Data.Raven;
+using HotChocolate.Data.Sorting.Expressions;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data;
+
+public static class RavenExecutableExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/RavenExecutableExtensions.cs:9]: summay

File: src/HotChocolate/Raven/src/Data/Sorting/Extensions/RavenSortingRequestExecutorBuilderExtensions.cs [hunk 15]
```diff
@@ -0,0 +1,14 @@
+using HotChocolate.Execution.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HotChocolate.Data;
+
+public static class RavenSortingRequestExecutorBuilderExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Sorting/Extensions/RavenSortingRequestExecutorBuilderExtensions.cs:6]: xml docs

File: src/HotChocolate/Raven/src/Data/Sorting/Extensions/RavenSortingSchemaBuilderExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,20 @@
+using HotChocolate.Data.Raven.Sorting;
+using HotChocolate.Data.Sorting;
+
+namespace HotChocolate.Data;
+
+public static class RavenSortingSchemaBuilderExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Sorting/Extensions/RavenSortingSchemaBuilderExtensions.cs:6]: xml docs

File: src/HotChocolate/Raven/src/Data/Sorting/RavenQueryableSortProvider.cs [hunk 17]
```diff
@@ -0,0 +1,30 @@
+using HotChocolate.Data.Sorting;
+using HotChocolate.Data.Sorting.Expressions;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+
+namespace HotChocolate.Data.Raven.Sorting;
+
+public class RavenQueryableSortProvider : QueryableSortProvider
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/src/Data/Sorting/RavenQueryableSortProvider.cs:9]: sealed?

docs

File: src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/FilterVisitorTestBase.cs [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+using HotChocolate.Types;
+using Microsoft.Extensions.DependencyInjection;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+using Squadron;
+
+namespace HotChocolate.Data.Filters;
+
+public abstract class FilterVisitorTestBase : IAsyncLifetime
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/FilterVisitorTestBase.cs:15]: docs

File: src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/__snapshots__/QueryableFilterVisitorComparableTests.Create_ShortIn_Expression.snap [hunk 19]
```diff
@@ -0,0 +1,81 @@
+12and13
+---------------
+{
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/__snapshots__/QueryableFilterVisitorComparableTests.Create_ShortIn_Expression.snap:3]: is this snapshot ok?

File: src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/__snapshots__/QueryableFilterVisitorComparableTests.Create_ShortNotIn_Expression.snap [hunk 20]
```diff
@@ -0,0 +1,81 @@
+12and13
+---------------
+{
+  "errors": [
```
Comment (michaelstaib) [inline src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/__snapshots__/QueryableFilterVisitorComparableTests.Create_ShortNotIn_Expression.snap:4]: is that snapshot ok?

File: src/HotChocolate/Core/src/Execution/DependencyInjection/SchemaRequestExecutorBuilderExtensions.Types.cs (hunk 1 above)
Comment (PascalSenn) [reply src/HotChocolate/Core/src/Execution/DependencyInjection/SchemaRequestExecutorBuilderExtensions.Types.cs:1281]: ah this one already exists. Removed

File: src/HotChocolate/Core/src/Types.OffsetPagination/Extensions/CursorPaginationResolverContextExtensions.cs (hunk 2 above)
Comment (PascalSenn) [reply src/HotChocolate/Core/src/Types.OffsetPagination/Extensions/CursorPaginationResolverContextExtensions.cs:6]: done

File: src/HotChocolate/Data/src/Data/Filters/Expressions/QueryableFilterProvider.cs (hunk 3 above)
Comment (PascalSenn) [reply src/HotChocolate/Data/src/Data/Filters/Expressions/QueryableFilterProvider.cs:136]: done

File: src/HotChocolate/Raven/src/Data/Pagination/RavenPagingContainer.cs (hunk 9 above)
Comment (PascalSenn) [reply src/HotChocolate/Raven/src/Data/Pagination/RavenPagingContainer.cs:126]: TaskHolder

File: src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/__snapshots__/QueryableFilterVisitorComparableTests.Create_ShortIn_Expression.snap (hunk 19 above)
Comment (PascalSenn) [reply src/HotChocolate/Raven/test/Data.Raven.Filters.Tests/__snapshots__/QueryableFilterVisitorComparableTests.Create_ShortIn_Expression.snap:3]: not needed

**Pull request closed without merging**